### ISSUE TITLE
Tokenizer

### ISSUE THREAD

**Issue by T-Atlas** [tloen/alpaca-lora#33]
Warning

```
The tokenizer class you load from this checkpoint is not the same type as the class this function is called from. It may result in unexpected tokenization. 
The tokenizer class you load from this checkpoint is 'LLaMATokenizer'. 
The class this function is called from is 'LlamaTokenizer'.
```

Caused by `tokenizer = LlamaTokenizer.from_pretrained("decapoda-research/llama-7b-hf", cache_dir="./cache/")`

Protential solution is modify `tokenizer = LLaMATokenizer.from_pretrained("decapoda-research/llama-7b-hf", cache_dir="./cache/")`
or modify`path_to_cache_hfmodel/models--decapoda-research--llama-7b-hf/snapshots/5f98eefcc80e437ef68d457ad7bf167c2c6a1348/tokenizer_config.json`

change 
`{"bos_token": "", "eos_token": "", "model_max_length": 1000000000000000019884624838656, "tokenizer_class": "LlamaTokenizer", "unk_token": ""}` to `{"bos_token": "", "eos_token": "", "model_max_length": 1000000000000000019884624838656, "tokenizer_class": "LLaMATokenizer", "unk_token": ""}`

**Comment by tloen** [comment]
Cc @zoidbb

**Comment by HideLord** [comment]
The cleanest solution would be to convert the original LLaMA weights to HF ones with the script included in the main Transformers branch, but it requires a fair bit of ram.

**Comment by vikigenius** [comment]
Yep, this leads to some tokenization artifacts and has started producing garbage outputs.
